### PR TITLE
feat: edit message menu option and handling its UI state [AR-2717]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2717" title="AR-2717" target="_blank">SubtaskAR-2717</a>  Add edit option for messages
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Currently we have "edit message" in message bottom sheet options disabled, we need a logic to handle edit message composer state correctly.

### Solutions

Added "edit" option for messages which changes the message composer to "edit message" state, handled closing this state, hiding the keyboard, back presses, requesting the focus, scrolling to the message that's being edited.

### Dependencies (Optional)

Needs releases with:

- https://github.com/wireapp/kalium/pull/1410

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Send a message and try to edit it by long-pressing and choosing "edit" option.

### Notes (Optional)

Currently it does not actually execute the edition, it's just the UI logic to change states and show everything properly.

### Attachments (Optional)

https://user-images.githubusercontent.com/30429749/215068690-32915530-cf36-48fa-9682-903c357b6156.mp4

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5587257)(5587257)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv0.0.1(5587257)(5587257)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5587477)(5587477)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv0.0.1(5587477)(5587477)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposerInnerState.kt [hunk 1]
```diff
@@ -418,7 +425,8 @@ sealed class MessageComposeInputState {
     val sendButtonEnabled: Boolean
         get() = this is Active && this.type is MessageComposeInputType.NewMessage && messageText.text.trim().isNotBlank()
     val editSaveButtonEnabled: Boolean
-        get() = this is Active && this.type is MessageComposeInputType.EditMessage && messageText.text.trim() != this.type.originalText
+        get() = this is Active && this.type is MessageComposeInputType.EditMessage && messageText.text.trim().isNotBlank()
+                && messageText.text.trim() != this.type.originalText
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposerInnerState.kt]: Shouldn't we trim originalText as well ?

**Review by ohassine — APPROVED** [review]
Well done ✨ just left a tiny comment about trimming the original text

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5620742)(5620742)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5620742)(5620742)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/8/) **failed**.

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5627491)(5627491)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5627491)(5627491)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1395/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.0(5627845)(5627845)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.0(5627845)(5627845)-staging-compat.apk)

**Pull request merged**